src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/NameApiTest.cs

```diff
@@ -0,0 +1,131 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Core.Modules.Dfs.Tests.Utils;
+using Catalyst.Core.Modules.Hashing;
+using MultiFormats.Registry;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Modules.Dfs.Tests.IntegrationTests.CoreApi
+{
+    public class NameApiTest
+    {
+        private IDfsService ipfs;
+        private ITestOutputHelper testOutput;
+
+        public NameApiTest(ITestOutputHelper output)
+        {
+            testOutput = output;
+            ipfs = TestDfs.GetTestDfs(output, null, null, new HashProvider(HashingAlgorithm.GetAlgorithmMetadata("sha2-256")));
+        }
+        
+        [Fact]
+        public async Task Resolve_Cid()
+        {
+            var actual = await ipfs.NameApi.ResolveAsync("QmYNQJoKGNHTpPxCBPh9KkDpaExgd2duMa3aF6ytMpHdao");
+            Assert.Equal("/ipfs/QmYNQJoKGNHTpPxCBPh9KkDpaExgd2duMa3aF6ytMpHdao", actual);
+
+            actual = await ipfs.NameApi.ResolveAsync("/ipfs/QmYNQJoKGNHTpPxCBPh9KkDpaExgd2duMa3aF6ytMpHdao");
+            Assert.Equal("/ipfs/QmYNQJoKGNHTpPxCBPh9KkDpaExgd2duMa3aF6ytMpHdao", actual);
+        }
+
+        [Fact]
+        public async Task Resolve_Cid_Path()
+        {
+            var temp = FileSystemApiTest.MakeTemp();
+            try
+            {
+                var dir = await ipfs.UnixFsApi.AddDirectoryAsync(temp);
+                var name = "/ipfs/" + dir.Id.Encode() + "/x/y/y.txt";
+                Assert.Equal("/ipfs/QmTwEE2eSyzcvUctxP2negypGDtj7DQDKVy8s3Rvp6y6Pc",
+                    await ipfs.NameApi.ResolveAsync(name));
+            }
+            finally
+            {
+                Directory.Delete(temp, true);
+            }
+        }
+        
+        [Fact]
+        public void Resolve_Cid_Invalid()
+        {
+            ExceptionAssert.Throws<FormatException>(() =>
+            {
+                var _ = ipfs.NameApi.ResolveAsync("QmHash").Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)